### ISSUE TITLE
Reference Browser Integration for Stateless DANE

### ISSUE THREAD

**Issue by rithvikvibhu** [opensystm/handshake-micro-grants#17]
### What have you built previously?

A bunch of Handshake-specific tools and services, listed at https://htools.work
All projects are open source: https://github.com/htools-org

### What will you be building? Why is that a valuable open source contribution to Handshake?

[Stateless DANE](https://hsd-dev.org/HIPs/proposals/0017/) is proposal that enables light clients to use minimal resources and can verify TLS trustlessly for any Handshake site without relying on the Handshake network all the time.

The goal here is to use the stateless dane library and integrate it with a browser: as a proof of concept, as a code demo to share with other browsers, and as a reference for alt-implementations.

**Rough Roadmap:**
- Set up a browser dev environment
- Build the browser without major changes
- Study where the code relevant to HTTPS and certificates are
- Add sane's build steps into browser's build process
- Modify UI to show hns info, settings to configure it (enable/disable)
- Figure out how the library fits into the request handling part (would have to be at a lower level, not a proxy like fingertip)
- Look into packaging all these changes for review / reference
- Write up on the changes made

To be clear, this is a reference, not a commitment to maintain a brand new browser, keeping it updated with security patches, etc.

**Similar works:**
- [Beacon Browser](https://github.com/imperviousinc/beacon) by Impervious includes DANE, but not Stateless DANE

### What are completion criteria?

- A basic integration of the stateless DANE library with a browser (chromium mostly)
- Document how it was done, and what other browsers need to do to get something similar working

### When will this be completed?

A bit hard to estimate, but once the stateless dane library is complete, ~4 weeks. Maybe a bit more based on the amount of work needed.

What is your contact information?

- Telegram `@rithvikvibhu`
- `rithvikvibhu @ gmail . com`

**Comment by Nathanwoodburn** [comment]
Would this work for mobile? Or is that more work?

**Comment by rithvikvibhu** [comment]
> Would this work for mobile? Or is that more work?

Anywhere. It's especially useful for mobile since unlike hnsd, it doesn't need to stay connected all the time.

**Comment by Nathanwoodburn** [comment]
Awesome. Any ideas how long it would take to convert your "demo" into a working android browser (as there isn't any yet)

**Comment by rithvikvibhu** [comment]
Idk about a browser, but with the stateless dane lib, it should be possible to build an external app (VPN-like proxy).

**Comment by Nathanwoodburn** [comment]
That sounds awesome.

**Comment by pcfreak30** [comment]
Kudos. This will be very valuable for Lume and HNS (and means I don't need to solve this myself :upside_down_face:)

**Comment by NetOpWibby** [comment]
HELL YEAH WOOOOO

**Comment by kauri-hero** [comment]
Congrats and amazing initiative to see - looking forward to it for [MASQ Browser](https://masqbrowser.com)
